### DISCUSSION TITLE
Add Rive animation tool compatibility inside Godot

### DISCUSSION THREAD

**Discussion by philippedcote** [godotengine/godot-proposals#5740] [Editor]
Hi everyone,

I love Godot but think its weakest point so far is the Animation Player / Timeline.

While waiting for this to improve, would it be possible to add Rive compatibility inside Godot?
https://rive.app

Thanks!

**Comment by Lielay9** [comment]
This is unfortunately something that will never be done in the core since Rive is a commercial product. Even if that wasn't the case, it'd likely be better to spend the time and effort improving the current implementation. That said, you can create a module to include the runtime [https://github.com/rive-app/rive-cpp](https://github.com/rive-app/rive-cpp) or commission it to be done by someone else... Or you can communicate the flaws the current implementation has so that they can be assessed and improved upon.

**Comment by awardell** [comment]
This seems like it might be a great use-case for a new GDExtension, no?

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/2924.
